tests/pass_security.rs

```diff
@@ -0,0 +1,114 @@
+pub mod common;
+use std::io::Error;
+use tokio::net::TcpStream;
+
+async fn read_from_server<'a>(buffer: &'a mut Vec<u8>, stream: &TcpStream) -> &'a str {
+    loop {
+        stream.readable().await.unwrap();
+        let n = match stream.try_read(buffer) {
+            Ok(n) => n,
+            Err(ref e) if e.kind() == std::io::ErrorKind::WouldBlock => {
+                continue;
+            }
+            Err(e) => panic!("{}", e),
+        };
+        return std::str::from_utf8(&buffer[0..n]).unwrap();
+    }
+}
+
+async fn send_to_server(buffer: &str, stream: &TcpStream) {
+    loop {
+        stream.writable().await.unwrap();
+
+        match stream.try_write(buffer.as_bytes()) {
+            Ok(_) => break,
+            Err(ref e) if e.kind() == std::io::ErrorKind::WouldBlock => {
+                continue;
+            }
+            Err(e) => panic!("{}", e),
+        };
+    }
+}
+
+async fn tcp_connect() -> Result<TcpStream, Error> {
+    let mut errcount: i32 = 0;
+    loop {
+        match TcpStream::connect("127.0.0.1:2121").await {
```

Comment:
We should probably not hardcode port 2121

Reply:
Not `2121` or not hardcode at all? We've hardcode it in all integration tests. I see no clean and simple way to avoid hardcoding it. Dynamic allocation I suppose would require a way to communicate the listening port between "threads".

Reply:
Replaced with `2150`

Reply:
It should probably be dynamic. But lets leave it for this PR.

Reply:
Yeah if we can get that sometime it would be nice.